### DISCUSSION TITLE
Does Vercel accept a value for stale-while-revalidate header?

### DISCUSSION THREAD

**Discussion by maximilianschmitt** [vercel/vercel#4688] [Help] [answered]
Hi!

I recently learned about stale-while-revalidate. In [this article](https://web.dev/stale-while-revalidate/) I learned that you can also supply a value in seconds, after which the CDN-cache should ALWAYS request a fresh version.

Does this work with Vercel?

For example, from my POV, the following code

```
Cache-Control: s-maxage=1, stale-while-revalidate=59
```

would mean: serve stale requests and revalidate in background every 1s. **If no user has accessed the resource for more than 59s, revalidate synchronously and serve the user an uncached version which also updates the cache**

Does it work like that with Vercel?

Thanks for your help!

Max

**Comment by mcsdevv** [comment, marked as answer]
Hello,

The Vercel platform does indeed accept a `stale-while-revalidate` header: https://vercel.com/docs/v2/edge-network/headers#cache-control-header

@shuding also created the SWR library which seems a worthwhile mention: https://github.com/vercel/swr

**Comment by maximilianschmitt** [comment]
Thanks @mcsdevv. I have successfully used `stale-while-revalidate` but I wasn't able to verify if setting a value, like `stale-while-revalidate=59` has any effect, it didn't seem so to me... 🤔

According to the [web.dev article](https://web.dev/stale-while-revalidate/), setting a parameter like `stale-while-revalidate=59` would mean that the cache forces a synchronous refresh after 59 seconds. If the Vercel CDN supported that, it could be a nice feature for lower traffic sites/pages.
